main/src/main/scala/sbt/Defaults.scala

```diff
@@ -2338,10 +2338,8 @@ object Classpaths {
         if (isSnapshot.value) "integration" else "release",
         ivyConfigurations.value.map(c => ConfigRef(c.name)).toVector,
         packagedArtifacts.in(publish).value.toVector,
-        checksums.in(publish).value.toVector, { //resolvername: not required if publishTo is false
-          val publishToOption = publishTo.value
-          if (publishArtifact.value) getPublishTo(publishToOption).name else "local"
-        },
+        checksums.in(publish).value.toVector,
+        getPublishTo(publishTo.value, publishArtifact.value).name, // error if publishTo is None and publishArtifact is true
```

Comment:
`publish / skip` might be better. See discussion here - https://github.com/sbt/sbt/issues/3136

Reply:
Well... I don't mind the new solutoin, but in any case the current code on lines 2342-2344 is clearly a bug and should be replaced soon with some sane alternative.

How about just merging this PR and then working the publish / skip into it as #3136 gets implemented.

Reply:
publish / skip was implemented here - https://github.com/sbt/sbt/pull/3380

Reply:
So you mean that instead of `publishArtifact.value` we should have here `(skip in publish).value` ?

Reply:
yes, it should be `!(skip in publish).value`, since you could have published artifacts even if unscoped `publishArtifact.value` returns false.

Reply:
actually could you hold off on that change until we understand what happened in the other PR please?

Reply:
I am wondering if a better fix would be to just revert https://github.com/sbt/sbt/pull/3760

Reply:
responded on https://github.com/sbt/sbt/pull/3760#issuecomment-518797685 - sounds like reverting that one is the way to go

Reply:
#3760 is what broke our build so if that is reverted, all is fine for me.

Reply:
@eed3si9n is it certain that if `skip in publish` is true, then nobody depends on the `publishConfiguration` ?

Reply:
publish, publishLocal, publishSigned, and sbt-bintray respect the skip in publish, so that should cover the relevant party.